### PR TITLE
(PUP-10730) Convert puppet http responses to net/http responses

### PR DESCRIPTION
**Pull request by joshcooper**
Previously puppet's legacy connection adapter assumed the http client implementation returned instances of `Puppet::HTTP::Response` that wrapped `Net::HTTPResponse`. But that isn't true when running in puppetserver.

Now the legacy connection adapter converts `Puppet::HTTP::Response` instances (returned by the new http client implementation) to `Net::HTTPResponse`. This makes it possible for report processors, rest termini and server functions that call the legacy http API to continue working in puppetserver 7.

### CODE REVIEW

**Comment by joshcooper** [comment]
https://github.com/puppetlabs/puppetdb/pull/3360 should be merged first

**Comment by joshcooper** [comment]
This needs to target `master`

File: lib/puppet/type/file/source.rb [hunk 1]
```diff
@@ -340,7 +340,7 @@ def chunk_file_from_source(&block)
 
     def handle_response_error(response)
       message = "Error #{response.code} on SERVER: #{response.body.empty? ? response.reason : response.body}"
-      raise Net::HTTPError.new(message, response.nethttp)
+      raise Net::HTTPError.new(message, Puppet::HTTP::ResponseConverter.to_ruby_response(response))
```
Comment (joshcooper) [inline lib/puppet/type/file/source.rb:343]: We really shouldn't be raising `Net::HTTPError` within puppet anymore, but that's a different issue than what this PR is fixing.

**Comment by joshcooper** [comment]
jenkins please test this with servertests

**Review by Magisus — APPROVED** [review]
I went through this a couple times, staged some questions, then read the commit messages and had my questions answered. So I think I'm 👍

**Pull request merged**